qcodes/instrument/parameter.py

```diff
@@ -1,39 +1,46 @@
 """
 Measured and/or controlled parameters
 
-The Parameter class is meant for direct parameters of instruments (ie
-subclasses of Instrument) but elsewhere in Qcodes we can use anything
-as a parameter if it has the right attributes:
-
-To use Parameters in data acquisition loops, they should have:
-    .name - like a variable name, ie no spaces or weird characters
-    .label - string to use as an axis label (optional, defaults to .name)
-    (except for composite measurements, see below)
-
-Controlled parameters should have a .set(value) method, which takes a single
-value to apply to this parameter. To use this parameter for sweeping, also
-connect its __getitem__ to SweepFixedValues as below.
-
-Measured parameters should have .get() which can return:
-
-- a single value:
-    parameter should have .name and optional .label as above
-
-- several values of different meaning (raw and measured, I and Q,
-  a set of fit parameters, that sort of thing, that all get measured/calculated
-  at once):
-    parameter should have .names and optional .labels, each a sequence with
-    the same length as returned by .get()
-
-- an array of values of one type:
-    parameter should have .name and optional .label as above, but also
-    .shape attribute, which is an integer (or tuple of integers) describing
-    the shape of the returned array (which must be fixed)
-    optionally also .setpoints, array(s) of setpoint values for this data
-    otherwise we will use integers from 0 in each direction as the setpoints
-
-- several arrays of values (all the same shape):
-    define .names (and .labels) AND .shape (and .setpoints)
+Anything that you want to either measure or control within QCoDeS should
+satisfy the Parameter interface. Most of the time that is easiest to do
+by either using or subclassing one of the classes defined here, but you can
+also use any class with the right attributes.
+
+TODO (alexcjohnson) update this with the real duck-typing requirements or
+create an ABC for Parameter and MultiParameter - or just remove this statement
+if everyone is happy to use these classes.
+
+This file defines four classes of parameters:
+
+``Parameter`` and ``MultiParameter`` must be subclassed:
+
+- ``Parameter`` is the base class for scalar-valued parameters, if you have
+    custom code to read or write a single value. Provides ``sweep`` and
+    ``__getitem__`` (slice notation) methods to use a settable parameter as
+    the swept variable in a ``Loop``. To use, fill in ``super().__init__``,
+    and provide a ``get`` method, a ``set`` method, or both.
+
```

Comment:
done in https://github.com/QCoDeS/Qcodes/pull/403/commits/4745d442754b46574c63f54a638072dc17a20807